frontend/app/routes/public/renew/$id/adult-child/children/$childId/information.tsx

```diff
@@ -103,6 +106,10 @@ export async function action({ context: { appContainer, session }, params, reque
       dateOfBirth: z.string(),
       clientNumber: z.string().trim().min(1, t('renew-adult-child:children.information.error-message.client-number-required')).refine(isValidClientNumberRenewal, t('renew-adult-child:children.information.error-message.client-number-valid')),
       isParent: z.boolean({ errorMap: () => ({ message: t('renew-adult-child:children.information.error-message.is-parent') }) }),
+      notFound: z
+        .boolean()
+        .optional()
+        .refine((val) => val === false || val === undefined, t('renew-adult-child:children.information.error-message.no-record')),
```

Comment:
No need to use Zod validation here, which in my opinion, should be reserved for field validation.

Instead, if there is no match, we can return a status from the `action(..)` and use `useFetcher()` to get that status to conditionally render the error message. 

This is currently done in `/applicant-information`: https://github.com/DTS-STN/canadian-dental-care-plan/blob/e01aa32f5216e5024adcf4966ec002fa2c1311b2/frontend/app/routes/public/renew/%24id/applicant-information.tsx#L142

Reply:
Changed to return status